tests/slsa_analyzer/checks/test_detect_malicious_metadata_check.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) 2024 - 2024, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+"""Module to test the malicious metadata detection check."""
+
+import json
+import os
+import urllib.parse
+from pathlib import Path
+
+import pytest
+from pytest_httpserver import HTTPServer
+
+from macaron.config.defaults import load_defaults
+from macaron.slsa_analyzer.build_tool.base_build_tool import BaseBuildTool
+from macaron.slsa_analyzer.checks.check_result import CheckResultType
+from macaron.slsa_analyzer.checks.detect_malicious_metadata_check import DetectMaliciousMetadataCheck
+from macaron.slsa_analyzer.package_registry.pypi_registry import PyPIRegistry
+from macaron.slsa_analyzer.specs.package_registry_spec import PackageRegistryInfo
+from tests.conftest import MockAnalyzeContext
+
+RESOURCE_PATH = Path(__file__).parent.joinpath("resources")
+
+
+@pytest.mark.parametrize(
+    ("purl", "expected"),
+    [
+        ("pkg:pypi/zlibxjson", CheckResultType.FAILED),
```

Comment:
Do you think it would be possible to test for the result of each heuristic of this package? I believe it would help us in knowing how the heuristic should behave.

Reply:
In additions, do you think it would make sense to also test for the scenario where the result is `CheckResultType.PASSED`?

Reply:
The `PASSED` case is already tested in the integration tests. Testing for `PASSED` requires uploading the resource files to enable offline testing. I could add a test case as I said, but it's also covered in the integration tests any way.

Reply:
Oh right, I forgot that the integration tests for PYPI packages already cover this. Thanks for reminding me.